pkg/prometheus/promcfg.go

```diff
@@ -750,8 +750,10 @@ func (cg *configGenerator) generateRemoteWriteConfig(version semver.Version, spe
 		if spec.WriteRelabelConfigs != nil {
 			relabelings := []yaml.MapSlice{}
 			for _, c := range spec.WriteRelabelConfigs {
-				relabeling := yaml.MapSlice{
-					{Key: "source_labels", Value: c.SourceLabels},
+				relabeling := yaml.MapSlice{}
+
+				if c.SourceLabels != "" {
```

Comment:
`.SourceLabels` is a slice of strings, not a single string, so this needs to be:


```suggestion
				if len(c.SourceLabels) > 0 {
```